### DISCUSSION TITLE
Which design pattern should I use?

### DISCUSSION THREAD

**Discussion by cvlnomen** [flame-engine/flame#647] [Q&A] [answered]
Today I wanted to start creating my first game using Flame. When looking at the official docs, I saw that the only structure that is required in a Flame project, is having the assets folder split up. I find this very strange, because Flutter apps can get really complex. Are there any design patterns that you guys can recommend to use with Flame?

**Comment by spydon** [comment, marked as answer]
Hi,
Flame is unopinionated when it comes to design patterns, just like Flutter.
Some design patterns which will very work well for large apps might be more overhead than needed for small apps etc.
It depends a lot on how your game will be designed too, if it is purely in Flame or if it is a mix of Flutter widgets and Flame, because if it is the latter you might have some state management in Flutter which will fit differently well with different design patterns.
Sorry for the vague answer, but there really is no silver bullet here.
